k8s_references.go

```diff
@@ -156,3 +157,29 @@ func (ks kobjSlice) process() ([]interface{}, error) {
 	}
 	return objectRefs, nil
 }
+
+// ResourceRef is a full Kubernetes resource reference composed of Group, Version, Kind, Namespace and Name.
+type ResourceRef struct {
```

Comment:
Can you please check if this can be named as `KResourceRef` now that there is no `func KResourceRef(....)`  ?

Reply:
It feels slightly strange that KObj and KRef are functions while this would be a struct, but I'm happy to go either way on this if there's consensus.

Reply:
For the sake of clarity and usability, I prefer just the struct (see [comment above](https://github.com/kubernetes/klog/pull/351#discussion_r971881010)) But for the sake of consistency we would have to use a function.

@serathius: what do you think?

Reply:
I agree with @pohly that struct should be preferred.